apps/next-evans/components/pages/home/news-feed/news-feed.ts

```diff
@@ -0,0 +1,507 @@
+    .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime())
+    .slice(0, MAX_NUM_NEWS_FEED_ITEMS);
+}
+
+function getRecentAdditions(
+  lang: Lang,
+  documents: Array<NewsfeedDocumentProps>,
+  altLanguageDocuments: Array<NewsfeedDocumentProps> = [],
+): FeedItem[] {
+  const type = LANG === `en` && lang === `es` ? `spanish_translation` : `book`;
+  return documents
+    .sort((a, b) => new Date(b.dateAdded).getTime() - new Date(a.dateAdded).getTime())
+    .map((doc) => {
+      return {
+        type,
+        title:
+          lang === `en`
+            ? htmlShortTitle(doc.title)
+            : `${htmlShortTitle(doc.title)}${LANG === `en` ? ` &mdash; (Spanish)` : ``}`,
+        ...dateFields(doc.mostModernEdition.impressionCreatedAt, LANG),
+        description:
+          type === `book`
+            ? LANG === `en`
+              ? `Download free eBook or pdf, or purchase a paperback at cost.`
+              : `Descárgalo en formato ebook o pdf, o compra el libro impreso a precio de costo.`
+            : `${doc.authorName}'s <em>${
+                altLanguageDocuments.find((altDoc) => altDoc.altLanguageId === doc.id)
+                  ?.title ?? ``
+              }</em> is now translated and available on the Spanish site.`,
+        url:
+          type === `spanish_translation`
+            ? `https://bibliotecadelosamigos.org/${doc.authorSlug}/${doc.slug}`
+            : `/${doc.authorSlug}/${doc.slug}`,
+      };
+    });
+}
+
+function getRecentAudios(
+  documents: Array<
+    Pick<
+      DocumentWithMeta,
+      'authorName' | 'dateAdded' | 'id' | 'title' | 'authorSlug' | 'slug' | 'editions'
+    >
+  >,
+): FeedItem[] {
+  return documents
+    .map((doc) =>
+      doc.editions.reduce<(Audiobook & { title: string; url: string }) | null>(
+        (acc, ed) => {
+          if (
+            ed.audiobook &&
+            new Date(ed.audiobook.dateAdded).getTime() >
+              new Date(acc ? acc.dateAdded : `January 1, 1970`).getTime()
+          ) {
+            return {
+              ...ed.audiobook,
+              title: htmlShortTitle(doc.title),
+              url: `${getDocumentUrl(doc.authorSlug, doc.slug)}#audiobook`,
+            };
+          }
+          return acc;
+        },
+        null,
+      ),
+    )
+    .filter(isNotNull)
+    .map((audio) => ({
+      type: `audiobook`,
+      title: `${audio.title} &mdash; (${LANG === `en` ? `Audiobook` : `Audiolibro`})`,
+      ...dateFields(audio.dateAdded, LANG),
+      description:
+        LANG === `en`
+          ? `Free audiobook is now available for download or listening online.`
+          : `El audiolibro ya está disponible para descargar gratuitamente o escuchar en línea.`,
+      url: audio.url,
+    }));
+}
+
+function dateFields(
+  dateStr: string,
+  lang: Lang,
+): Pick<FeedItem, 'month' | 'year' | 'day' | 'date'> {
+  const date = new Date(dateStr);
+  const month = months[lang][date.getMonth()]?.substring(0, 3) || ``;
```

Comment:
because dates are weird and hard and a source of lots of bugs, i think i'd feel better about using `invariant` and getting a build-time error rather than falling back to the empty string here, b/c i think there's a non-trivial chance it might happen.